nixos/modules/services/x11/extra-layouts.nix

```diff
@@ -0,0 +1,154 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  layouts = config.services.xserver.extraLayouts;
+
+  layoutOpts = {
+    options = {
+      description = mkOption {
+        type = types.str;
+        description = "A short description of the layout.";
+      };
+
+      languages = mkOption {
+        type = types.listOf types.str;
+        description =
+        ''
+          A list of languages provided by the layout.
+          (Use ISO 639-2 codes, for example: "eng" for english)
+        '';
+      };
+
+      filepath = mkOption {
+        type = types.path;
+        description = "The path to the layout file.";
+      };
+    };
+  };
+
+  patchIn = name: layout:
+  ''
+      # install layout file
+      cp ${layout.filepath} symbols/${name}
+
+      # patch makefiles
+      ${pkgs.ed}/bin/ed -v symbols/Makefile.am <<EOF
+      /symbols_DATA =
+      a
+      ${name} \\
+      .
+      w
+      EOF
+      ${pkgs.ed}/bin/ed -v symbols/Makefile.in <<EOF
+      /symbols_DATA =
+      a
+      ${name} \\
+      .
+      w
+      EOF
+
+      # add layout description
+      ${pkgs.ed}/bin/ed -v rules/base.xml.in <<EOF
+      /<\/layoutList>
+      -
+      a
+      <layout>
+        <configItem>
+          <name>${name}</name>
+          <_shortDescription>${name}</_shortDescription>
+          <_description>${layout.description}</_description>
+          <languageList>
+            ${lib.concatMapStrings (lang: "<iso639Id>${lang}</iso639Id>\n") layout.languages}
+          </languageList>
+        </configItem>
+        <variantList/>
+      </layout>
+      .
+      w
+      EOF
+      cp rules/base.xml.in rules/base.evdev.xml.in
+  '';
+
+  xkb_patched = pkgs.xorg.xkeyboardconfig.overrideAttrs (old: {
```

Comment:
I just noticed this, you're defining `xkb_patched` in terms of `pkgs`, but then use `xkb_patched` in an overlay below, but `pkgs` needs to evaluate all overlays to evaluate, so I think this means it needs to evaluate nixpkgs twice to get the results. You should define `xkb_patched` within the overlay (in terms of `self`) instead

Reply:
Wouldn't hurt to also export the patched version as an attribute in the overlay.

Reply:
Uhm, does `patchIn`, which needs `pkgs.ed`, has to be moved in the overlay as well?

Reply:
Ok, done.